### PR TITLE
Add support for pre-uploaded blobs

### PR DESCRIPTION
**Pull request by gabrielBusta**
This PR adds support for pre-uploaded blobs (so we can easily use the blobs uploaded to the archive by the addon pipeline automation)

### CODE REVIEW

File: morgoth/cli.py [hunk 1]
```diff
@@ -147,54 +147,56 @@ def make_release(xpi_file, bearer, profile, verbose, reupload):
...
-                index = 2
-                check_path = xpi.get_ftp_path(prefix, suffix='-{}'.format(index))
-                while check_path in existing_files:
-                    if validate_uploaded_xpi_hash(xpi, bucket, check_path):
-                        uploaded = True
-                        suffix = '-{}'.format(index)
-                        break
-                    index += 1
-                    check_path = xpi.get_ftp_path(prefix, suffix='-{}'.format(index))
+        if not xpi.archived:
```
Comment (gabrielBusta) [inline morgoth/cli.py:150]: The diff looks convoluted... but the logic remains the same. I just enclosed the logic that uploads the file as is within the `if not xpi.archived` block

File: morgoth/cli.py [hunk 2]
```diff
@@ -147,54 +147,56 @@ def make_release(xpi_file, bearer, profile, verbose, reupload):
...
+                            index += 1
+                            suffix = '-{}'.format(index)
+                            upload_path = xpi.get_ftp_path(prefix, suffix=suffix)
+                with open(xpi.path, 'rb') as data:
+                    bucket.put_object(Key=upload_path, Body=data)
+                    output('XPI uploaded to: {}'.format(upload_path), Fore.GREEN)
+            release_data = xpi.generate_release_data(
+                base_url=settings.get('aws.base_url', DEFAULT_AWS_BASE_URL), prefix=prefix, suffix=suffix)
+        else:
+            release_data = xpi.generate_release_data()
```
Comment (gabrielBusta) [inline morgoth/cli.py:199]: When we use a pre-uploaded blob the url doesn't need to be generated.

File: morgoth/xpi.py [hunk 3]
```diff
@@ -57,8 +59,21 @@ class BadXPIfile(Exception):
         pass
 
     def __init__(self, path):
-        if not os.path.isfile(path):
-            raise XPI.DoesNotExist()
+        if path.startswith("https://") or path.startswith("http://"):
+            response = requests.get(path)
```
Comment (escapewindow) [inline morgoth/xpi.py:63]: We could add a retry here, e.g. [redo](https://pypi.org/project/redo/) if that makes sense to do.

**Review by escapewindow — APPROVED** [review]
Looks good, thank you!

**Pull request merged**